server/api/src/rsDataObjUnlink.cpp

```diff
@@ -198,32 +248,30 @@ rsDataObjUnlink( rsComm_t *rsComm, dataObjInp_t *dataObjUnlinkInp ) {
     return status;
 }
 
-int
-_rsDataObjUnlink( rsComm_t *rsComm, dataObjInp_t *dataObjUnlinkInp,
-                  dataObjInfo_t **dataObjInfoHead ) {
-    int status;
-    int retVal = 0;
+int _rsDataObjUnlink(rsComm_t *rsComm,
+                     dataObjInp_t *dataObjUnlinkInp,
+                     dataObjInfo_t **dataObjInfoHead)
+{
     dataObjInfo_t *tmpDataObjInfo, *myDataObjInfoHead;
-
-    status = chkPreProcDeleteRule( rsComm, dataObjUnlinkInp, *dataObjInfoHead );
+    int status = chkPreProcDeleteRule( rsComm, dataObjUnlinkInp, *dataObjInfoHead );
     if ( status < 0 ) {
         return status;
     }
 
-
     myDataObjInfoHead = *dataObjInfoHead;
     if ( strstr( myDataObjInfoHead->dataType, BUNDLE_STR ) != NULL ) { // JMC - backport 4658
-        int numSubfiles; // JMC - backport 4552
+        // TODO Shouldn't this be "clientUser" instead of "proxyUser"?
```

Comment:
@jasoncoposky @alanking I'm 0-for-1000 on getting this right.   Thoughts?

Reply:
This could be correct. I grep'd the codebase and found several other places that do a similar check of the proxy user. In my experience, I've always modified or checked the `clientUser` info depending on what needed to be done. This just had me wondering when should the proxy user be inspected? Always? Should it be checked whenever the client user's info is checked?  It is not clear to me. Is the proxy user just for making the connection to other servers?